### ISSUE TITLE
numerical stability in interp_2d

### ISSUE THREAD

**Issue by synikitin** [tnagler/kdecopula#2]
Hello,

When using kdevine, I observed an odd effect of order of observations on density calculation. I traced it in a debugger to interp_2d function. Here is a pattern of behavior:

```
Browse[5]> u
             zr2       zr1
[1,] 0.001724833 0.0010000
[2,] 0.858983907 0.7091684

Browse[5]> interp_2d(u[1, , drop = F], obj$estimate, obj$grid, numeric(4), 
+           numeric(4))
[1] 161.8257
Browse[5]> interp_2d(u[2, , drop = F], obj$estimate, obj$grid, numeric(4), 
+           numeric(4))
[1] 1.141078

Browse[5]> interp_2d(u[c(1, 1), , drop = F], obj$estimate, obj$grid, numeric(4), 
+           numeric(4))
[1] 161.8257 161.8257
Browse[5]> interp_2d(u[c(2, 2), , drop = F], obj$estimate, obj$grid, numeric(4), 
+           numeric(4))
[1] 1.141078 1.141078
Browse[5]> interp_2d(u[c(1, 2), , drop = F], obj$estimate, obj$grid, numeric(4), 
+           numeric(4))
[1] 161.825726   1.141078
Browse[5]> interp_2d(u[c(2, 1), , drop = F], obj$estimate, obj$grid, numeric(4), 
+           numeric(4))
[1] 1.141078e+00 1.000588e-15

```

Note the effect of switching from (1, 2) to (2, 1). Overall, it appears that having two observations in a particular order can crash the function.
 
I glanced at the interp_2d function, but that leads to other functions, which may take me a while to figure out. Do you have any thoughts on how to stabilize the calculation?

Thanks,

Slava

**Comment by tnagler** [comment]
Hi, it's probably related to not resetting `i` and `j` to 0 in the loop over `n`. I will look into this, thanks for making me aware!

**Comment by tnagler** [comment]
Fixed by #3, new version will be on CRAN in 1-2 days. Thanks again!
